File: backend/src/main/java/com/noisevisionsoftware/nutrilog/model/diet/Diet.java
package com.noisevisionsoftware.nutrilog.model.diet;

import com.google.cloud.Timestamp;
import lombok.Builder;
import lombok.Data;
import lombok.Setter;

import java.util.List;

@Data
@Setter
@Builder
public class Diet {
    private String id;
    private String userId;
    private Timestamp createdAt;
    private Timestamp updatedAt;
    private List<Day> days;
    private DietMetadata metadata;
}

File: app/src/main/java/com/noisevisionsoftware/vitema/domain/model/health/dietPlan/Diet.kt
package com.noisevisionsoftware.vitema.domain.model.health.dietPlan

import com.google.firebase.Timestamp
import com.google.firebase.firestore.PropertyName
import com.google.gson.annotations.JsonAdapter
import com.noisevisionsoftware.vitema.utils.TimestampAdapter

data class Diet(
    val id: String = "",
    val userId: String = "",

    @JsonAdapter(TimestampAdapter::class)
    @PropertyName("createdAt")
    val createdAt: Timestamp = Timestamp.now(),

    @JsonAdapter(TimestampAdapter::class)
    @PropertyName("updatedAt")
    val updatedAt: Timestamp = Timestamp.now(),

    val days: List<DietDay> = emptyList(),
    val metadata: DietMetadata = DietMetadata()
)